### PR TITLE
[Dev] Update storybooks to v3

### PR DESCRIPTION
**Pull request by orta**
This PR will update storybooks to 3.x - it's not there yet because I ended up with some typescript errors from the code, which was surprising.

This updates the webpack server in Storybooks to v2, which is why there's config changes

### CODE REVIEW

**Comment by alloy** [comment]
What are the errors?

**Comment by orta** [comment]
screen shot 2017-06-20 at 18 47 47

Looks like this:

```
$ "/Users/orta/dev/projects/artsy/js/libs/reaction-force/node_modules/.bin/tsc" --noEmit
src/components/artsy.tsx(91,25): error TS2698: Spread types may only be created from object types.
src/components/input.tsx(98,20): error TS2698: Spread types may only be created from object types.
src/components/inquiry_artwork/index.tsx(38,7): error TS6133: 'Overlay' is declared but never used.
src/components/inquiry_artwork/index.tsx(49,7): error TS2698: Spread types may only be created from object types.
src/components/message.tsx(12,10): error TS2698: Spread types may only be created from object types.
src/components/text_area.tsx(14,20): error TS2322: Type '{ error?: boolean; block?: boolean; defaultChecked?: boolean; defaultValue?: string | string[]; s...' is not assignable to type 'ChangeTargetHTMLProps<HTMLTextAreaElement>'.
  Property 'error' does not exist on type 'ChangeTargetHTMLProps<HTMLTextAreaElement>'.
src/components/title.tsx(25,10): error TS2322: Type '{ titleSize?: TitleSize; defaultChecked?: boolean; defaultValue?: string | string[]; suppressCont...' is not assignable to type 'HTMLProps<HTMLDivElement>'.
  Property 'titleSize' does not exist on type 'HTMLProps<HTMLDivElement>'.

```

**Comment by orta** [comment]
This was master, which is weird, cause I saw this working on @eessex's I'm sure

**Comment by orta** [comment]
I re-cloned, it went away

File: webpack/development.js [hunk 1]
```diff
@@ -30,5 +30,6 @@ module.exports = () => {
       new CheckerPlugin(),
       new webpack.HotModuleReplacementPlugin(),
     ],
+    output: null,
```
Comment (orta) [inline webpack/development.js]: nope

**Comment by orta** [comment]
OK, this is working 👍

**Comment by orta** [comment]
screen shot 2017-06-21 at 15 24 00


If anyone know what asset optimisation is for web pack, we can reduce the time in HMR is by tweaking that

screen shot 2017-06-21 at 15 44 02

**Comment by damassi** [comment]
@orta - Can you swap out this "STORYBOOK" button url and label to read something like "REACTION FORCE" or "HOME" or something? Its super annoying to click it and be taken away from the app, when it should really link to https://artsy.github.io/reaction-force/.

screen shot 2017-06-21 at 12 52 42 pm

**Comment by orta** [comment]
hahah you can finally get that feature @damassi - doing it w

**Comment by damassi** [comment]
screen shot 2017-06-21 at 12 58 31 pm

File: .storybook/webpack.config.js [hunk 2]
```diff
@@ -1,30 +1,48 @@
...
+const sharifyPath = sharify({ METAPHYSICS_ENDPOINT });
+
+// A mix of  the base from Emission's webpack setup, and the simple config for
+// storybooks: https://storybook.js.org/configurations/custom-webpack-config/
 
 module.exports = {
   devtool: "#inline-source-map", // Otherwise getting errors about e.g. `Relay` not being defined.
   resolve: {
-    extensions: ["", ".js", ".jsx", ".ts", ".tsx"],
+    extensions: [".js", ".jsx", ".ts", ".tsx"],
```
Comment (broskoski) [inline .storybook/webpack.config.js]: why remove the empty extension?

Comment (orta) [inline .storybook/webpack.config.js]: Webpack 2 errors on load for that specifically, think whatever that implied is now handled implicitly

Comment (broskoski) [inline .storybook/webpack.config.js]: 👍

**Comment by broskoski** [comment]
LGTM, except for the merge conflicts!

File: package.json [hunk 3]
```diff
@@ -91,6 +89,8 @@
     "webpack-stream": "^3.2.0"
   },
   "dependencies": {
+    "@storybook/addon-options": "^3.1.2",
+    "@storybook/react": "^3.1.3",
```
Comment (alloy) [inline package.json]: @orta I’m wondering if it was intentional to add these as runtime dependencies? I’m going to assume not and will move them to dev deps.

**Pull request merged**